src/client/src/App/Credit/SellSide/window.ts

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from "react"
+import { RFQ_ID_PARAM } from "@/utils"
+import { highlightRfqId } from "./sellSideState"
+
+const getRfqSearchParam = (url: string) =>
+  url?.split("?")?.[1]?.split(`${RFQ_ID_PARAM}=`)?.[1]
+
+const highlightNewRfq = (url: string) => {
+  const rfqId = getRfqSearchParam(url)
+  return highlightRfqId(rfqId == null ? null : Number(rfqId))
+}
+
+/**
+ * Reacts to rfq id in route after history changes to update
+ * sell-side event streams
+ */
+export const useRfqSearchParamEffect = () => {
+  useEffect(() => {
+    // Intercepts pushState and selects rfqId to highlight
+    window.history.pushState = new Proxy(window.history.pushState, {
```

Comment:
ooh, fancy .. second time in my life I think that I have seen ES Proxy used 👍🏻

Reply:
Yeah, I found this example online.  It can function as a "listener" on method calls to run side effects.